docs/getting-started-tutorial/4-css-selector-transforms.adoc

```diff
@@ -0,0 +1,171 @@
+:idprefix:
+:idseparator: -
+:toc: right
+:toclevels: 2
+
+# CSS Selector Transforms
+
+Because Lift operates by transforming HTML trees, it needs an easy way to
+specify those transformations. Otherwise we'd be doing a bunch of recursive
+tree searches and munges and it would get ugly and probably slow to boot. To
+deal with transformations easily, we use a small subset of CSS selectors, with
+a few Lift idiosyncrasies to maximize performance and address some use cases
+that are particularly useful when transforming trees.
+
+We'll leave forms for the next section, as forms always come with a catalog of
+related functionality, and focus on binding the list of chat messages in this
+section. We'll also add a new one before every page load, so that we can see
+the list changing.
+
+First, we'll define a variable to hold the messages:
+
+```
+...
+object Chat {
+  var messageEntries = List[String]()
+...
+}
+```
+
+Then, we can change the definition of the `messages` method to bind the
+contents of the message list:
+
+```
+...
+
+import net.liftweb.util.Helpers._
+
+...
+  def messages = {
+    "li *" #> messageEntries
+  }
```

Comment:
Examples in this form are frequently confusing to me when reading getting started guides. Can we present the entire `Chat` object with imports in each of these since it's fairly self contains?

Reply:
I actually want the compiled output here to produce the whole thing with highlighted changes… I have some more thoughts here. Bit off a little more than I could chew the first time I tried to do this, but I think there are some small things we can do to get to that point quickly.